doc/source/programs/gdaltindex.rst

```diff
@@ -16,23 +16,73 @@ Synopsis
 .. code-block::
 
     gdaltindex [--help] [--help-general]
+            [-overwrite] [-recursive] [-filename_filter <val>]...
+            [-min_pixel_size <val>] [-max_pixel_size <val>]
             [-f <format>] [-tileindex <field_name>] [-write_absolute_path]
             [-skip_different_projection] [-t_srs <target_srs>]
             [-src_srs_name <field_name>] [-src_srs_format {AUTO|WKT|EPSG|PROJ}]
-            [-lyr_name <name>] <index_file> <gdal_file> [<gdal_file>]...
+            [-lyr_name <name>]
+            [-gti_filename <name>]
+            [-tr <xres> <yres>] [-te <xmin> <ymin> <xmax> <ymax>]
+            [-ot <datatype>] [-bandcount <val>] [-nodata <val>[,<val>...]]
+            [-colorinterp <val>[,<val>...]] [-mask]
+            [-mo <KEY>=<VALUE>]...
+            [-fetch_md <gdal_md_name> <fld_name> <fld_type>]...
+            <index_file> <file_or_dir> [<file_or_dir>]...
 
 Description
 -----------
 
 This program creates an OGR-supported dataset with a record for each input raster file,
 an attribute containing the filename, and a polygon geometry outlining the
 raster.  This output is suitable for use with `MapServer <http://mapserver.org/>`__ as a raster
-tileindex.
+tileindex, or as input for the :ref:`GTI <raster.gti>` driver.
 
-.. program:: ogrtindex
+.. program:: gdaltindex
 
 .. include:: options/help_and_help_general.rst
 
+.. option:: -overwrite
+
+    .. versionadded:: 3.9
+
+    Overwrite the tile index if it already exists.
+
+.. option:: -recursive
+
+    .. versionadded:: 3.9
+
+    Whether directories specified in <file_or_dir> should be explored recursively.
+
+.. option:: -filename_filter <val>
+
+    .. versionadded:: 3.9
+
+    Pattern that the filenames contained in directories pointed by <file_or_dir>
+    should follow.
+    '*' is a wildcard character that matches any number of any characters
+    including none. '?' is a wildcard character that matches a single character.
+    Comparisons are done in a case insensitive way.
+    Several filters may be specified
+
+    For example :``-filename_filter "*.tif" -filename_filter "*.tiff"``
+
+.. option:: -min_pixel_size <val>
```

Comment:
Of course I may have misunderstood the purpose of this option, if that's the case @rouault will safely ignore my comment.

Reply:
changed text to "Minimum pixel size in term of geospatial extent per pixel (resolution)".  The thing is that speaking about minimum resolution is always a bit confusing, because a raster with 1 meter/pixel is better resoluted than one with 10 meter/pixel , but 1 < 10 ...